src/building/count.c

```diff
@@ -59,6 +147,195 @@ int building_count_upgraded(building_type type)
     return upgraded;
 }
 
+int building_count_in_area(building_type type, int minx, int miny, int maxx, int maxy)
+{
+    int array_size = building_count();
+    int last_entry_position = 0;
+    int* found_buildings = (int*) malloc(array_size * sizeof(int));
```

Comment:
You don't need to alloc such a large memory array.

You know that at most you'll have as many buildings as the area to check.

So, you can set the `array_size` as:

```c 
int grid_area = abs((maxx - minx) * (maxy - miny));
int array_size = grid_area < building_count() ? grid_area : building_count();
```

Also, nitpick. Code format should be:

```c
    int *found_buildings = (int *) malloc(array_size * sizeof(int));
```

Reply:
Very good point. I should have thought of that.

Reply:
Fixed.